devel-tools/perf-benchmark

```diff
@@ -54,54 +70,77 @@ while (( "$#" )); do
 	  ZEEK_BUILD=$2
 	  shift 2
 	  ;;
+      -m|--mode)
+	  MODE=$2
+	  shift 2
+	  ;;
       -i|--interface)
 	  INTERFACE=$2
 	  shift 2
 	  ;;
       -f|--flamegraph)
-	  MODE="flamegraph"
 	  FLAMEGRAPH_PATH=$2
-	  if [ -z $FLAMEGRAPH_OUTPUT ]; then
-	      FLAMEGRAPH_OUTPUT="benchmark.svg"
-	  fi
 	  shift 2
 	  ;;
       -o|--output)
-	  FLAMEGRAPH_OUTPUT=$2
+	  FLAMEGRAPH_PREFIX=$2
+	  shift 2
+	  ;;
+      -s|--seed)
+	  SEED_FILE=$2
 	  shift 2
 	  ;;
   esac
 done
 
-if [ -z $ZEEK_BUILD ]; then
+if [ "${MODE}" != "intf" -a "${MODE}" != "file" -a "${MODE}" != "flamegraph" ]; then
+    echo "Error: -m argument should be one of 'intf', 'file', or 'flamegraph'"
+    echo
+    usage
+fi
+
+if [ -z "${ZEEK_BUILD}" ]; then
     echo "Error: -b argument is required and should point at a Zeek binary"
     echo
     usage
 fi
 
-if [ -z $DATA_FILE ]; then
+if [ -z "${DATA_FILE}" ]; then
     echo "Error: -d argument is required and should point at a pcap file to replay"
     echo
     usage
 fi
 
-if [ -z $INTERFACE ]; then
-    echo "Error: -i argument is required and should point to an idle network interface"
+if [ "${MODE}" != "file" -a -z "${INTERFACE}" ]; then
+    echo "Error: -i argument is required for the ${MODE} mode and should point to an idle network interface"
     echo
     usage
 fi
 
 # Various run-time options
-ZEEK_ARGS="-i af_packet::${INTERFACE}"
 ZEEK_CPU=10
 TCPREPLAY_CPU=11
 
-echo "Running '${ZEEK_BUILD} ${ZEEK_ARGS}' against ${DATA_FILE}"
+ZEEK_ARGS=""
+if [ "${MODE}" != "file" ]; then
+    ZEEK_ARGS="-i af_packet::${INTERFACE}"
+fi
+
+if [ -n "${SEED_FILE}" ]; then
+    ZEEK_ARGS="${ZEEK_ARGS} -G ${SEED_FILE}"
+fi
+
+PRELOAD_LIBS=`for i in $(ls -1L /home/tim/3rdparty/libs/*); do echo -n "$(readlink -f $i) "; done`
```

Comment:
This path-hardcoding probably not meant to be included?

Then also the test right below probably means to check for non-zero string length w/ `-n`.

Reply:
Aw damn, I meant to remove that before I PR'd and forgot it. I'll take it out.

Reply:
Fixed.